components/calendar/calendar.js

```diff
@@ -663,87 +707,122 @@ class Calendar extends LocalizeStaticMixin(RtlMixin(LitElement)) {
 			e.preventDefault();
 			e.stopPropagation();
 		}
+
+		if (numDaysChange) {
+			const oldFocusDate = new Date(this._focusDate);
+			const possibleFocusDate = new Date(
+				this._focusDate.getFullYear(),
+				this._focusDate.getMonth(),
+				this._focusDate.getDate() + numDaysChange
+			);
+
+			// if HOME or END _focusDate becomes earliest or latest non-disabled date in the week as applicable
+			// if arrow keys change _focusDate only if intended target is not disabled
+			if (e.keyCode === keyCodes.END) this._updateFocusDateDependentOnDisabled(possibleFocusDate, true);
+			else if (e.keyCode === keyCodes.HOME) this._updateFocusDateDependentOnDisabled(possibleFocusDate);
+			else if (getDisabled(possibleFocusDate, this.minValue, this.maxValue)) return;
+			else this._focusDate = possibleFocusDate;
+
+			this._isInitialFocusDate = false;
+
+			const date = await this._getDateElement(this._focusDate);
+			if (!date) {
+				this._keyboardTriggeredMonthChange = true;
+				const upDown = Math.abs(numDaysChange) !== 1; // use left/right animation if difference is 1 day
+				if (oldFocusDate < this._focusDate) {
+					this._updateShownMonthIncrease(true, upDown);
+				} else {
+					this._updateShownMonthDecrease(true, upDown);
+				}
+			} else {
+				this._focusDateAddFocus();
+			}
+			await this.updateComplete;
+			this._isInitialFocusDate = true;
+		}
 	}
 
 	_onNextMonthButtonClick() {
 		this._updateShownMonthIncrease();
-		this._updateFocusDate();
+		this._updateFocusDateOnMonthButtonClick();
 	}
 
 	_onPrevMonthButtonClick() {
 		this._updateShownMonthDecrease();
-		this._updateFocusDate();
+		this._updateFocusDateOnMonthButtonClick();
 	}
 
-	async _updateFocusDate() {
+	async _updateFocusDateOnMonthButtonClick() {
 		const selectedValueDate = this.selectedValue ? getDateFromISODate(this.selectedValue) : null;
 		const dateElem = selectedValueDate ? await this._getDateElement(selectedValueDate) : null;
-		if (dateElem) {
+		if (dateElem && !getDisabled(selectedValueDate, this.minValue, this.maxValue)) {
 			this._focusDate = selectedValueDate;
 		} else {
-			this._focusDate = new Date(this._shownYear, this._shownMonth, 1);
+			this._updateFocusDateDependentOnDisabled(new Date (this._shownYear, this._shownMonth, 1));
 		}
 	}
 
-	async _updateFocusDateOnKeyDown(numDays) {
-		const oldFocusDate = new Date(this._focusDate);
-		this._focusDate = new Date(
-			this._focusDate.getFullYear(),
-			this._focusDate.getMonth(),
-			this._focusDate.getDate() + numDays
-		);
-		const date = await this._getDateElement(this._focusDate);
-		if (!date) {
-			this._keyboardTriggeredMonthChange = true;
-			const upDown = Math.abs(numDays) !== 1; // use left/right animation if difference is 1 day
-			if (oldFocusDate < this._focusDate) {
-				this._updateShownMonthIncrease(true, upDown);
-			} else {
-				this._updateShownMonthDecrease(true, upDown);
-			}
+	_updateFocusDateDependentOnDisabled(possibleFocusDate, latestPossibleFocusDate) {
```

Comment:
1. If desired focus date is between min and max, it becomes focus date
2. If there are dates in the month that are not disabled, the focus date is either the earliest or latest depending on if `latestPossibleFocusDate` is true
3. If there are no enabled dates, focus date is undefined.